pkg/controller/common/annotation/controller_version.go

```diff
@@ -126,37 +116,30 @@ func ReconcileCompatibility(ctx context.Context, client k8s.Client, obj ctrlclie
 	}
 
 	// if the current version is gte the minimum version then they are compatible
-	if currentVersion.GTE(minVersion) {
```

Comment:
Not sure why we parse the controller version here and then only use it for logging. (Nothing you changed)  https://github.com/elastic/cloud-on-k8s/pull/4197/files#diff-b45319baa705e2377f984ec51682b15436fdf75a4db9c04e604c443c4b200546L123-L125

Reply:
I _think_ it is to throw an early error. We have [some unit tests](https://github.com/elastic/cloud-on-k8s/blob/a0fe4502ac74242e66d92d5af1f90db952b32cc7/pkg/controller/association/reconciler_test.go#L275) which are expecting an error to be raised if the controller version can't be parsed.

Reply:
The unit test I mentioned is odd, its result does not depend on the value of the annotation as it should be:

```golang
func TestReconciler_Reconcile_NotCompatible(t *testing.T) {
	kb := sampleKibanaWithESRef()
	// set an incompatible controller annotation
	kb.Annotations = map[string]string{
		annotation.ControllerVersionAnnotation: "0.9.0",
	}
	r := testReconciler(&kb)
	_, err := r.Reconcile(context.Background(), reconcile.Request{NamespacedName: k8s.ExtractNamespacedName(&kb)})
	// should error out
	require.Error(t, err)
}
```

If the controller version in the annotation is not compatible we should not expect an error, the resource should be skipped.
When running this test the error we have is `No Major.Minor.Patch elements found`, it's because the controller version used in the test reconciler is wrong (`BuildInfo: about.BuildInfo{Version: "unit-tests"}`).
Setting a compatible version in the annotation still results in an error, it should not be the case.
I will raise a separate issue to investigate later.